bots/README.md

```diff
@@ -1,41 +1,69 @@
+the collection of messages for their user account, creating a stream that is updated every time a message is sent either directly to them or any room they are joined in.
+
+For more details, see the [Bots Architecture section](bots-architecture/).
+
+## Bots types
+
+Depending on where bot's logic is defined, bots can be internal and external.
+
+**Internal** bots run as an app attached to the instance and are tightly coupled with
+the architecture of Rocket.Chat. Generally, this type of bots has limited features
+for conversational UI but can provide deeper integration with the messaging platform.
+Internal bots can be used to provide simple command driven interfaces for Rocket.Chat users.
+
+**External** bots are hosted outside of your Rocket.Chat instance and their working logic
+is defined by their provider. External bots are best suited to heavy messaging interactions and
+setups where maintenance or processes are independent of Rocket.Chat.
+
+### Bots types support in Rocket.Chat
+
+Rocket.Chat supports for multiple bot frameworks and platforms, including:
+[bBot](http://bbot.chat/), [Hubot](https://hubot.github.com/), [Botkit](https://botkit.ai/),
+[Botpress](https://botpress.io/).
+
+## How are bots hosted?
+
+The hosting for **external** bots depends on the platform. For example, Hubot can be
+launched on any Node.js environment, but you have to set it up yourself (using Heroku or
+Glitch services). Another bot platforms like Botkit provide their own services for hosting and provisioning bots. They just need a configuration to connect to your Rocket.Chat instance.
+
+**Internal** bots are hosted as part of the Rocket.Chat instance, either on-premise
+or in the same cloud container. Their provisioning and operation is dependent on the
+Rocket.Chat server.
 
 ## Getting Started
 
-These are the basic steps for using bots with Rocket.Chat.
+These are the basic steps for using bots with Rocket.Chat:
+
+1. A bot user is [created by an admin](creating-bot-users/) on the server;
+2. The bot is running as [a separate process](bots-architecture/) using your chosen framework or platform;
+3. The bot environment is pre-configured with [environment variables](configure-bot-environment/);
+4. The bot's behaviour is defined via scripts according to the requirements of your framework. For example, check the details on how to [run a bBot bot](running-a-bbot-bot/).
 
-1. A bot user must be [created by an admin](creating-bot-users/) on the server
-2. The bot is run as [a separate process](bots-architecture/) using your chosen framework or platform
-3. The bot environment must be pre-configured by [setting environment variables](configure-bot-environment/)
-4. You provide scripts to define the bot's behaviour according to the requirements of your framework
-    - e.g. See [Running a bBot bot](running-a-bbot-bot/)
+## Next steps
+
+Get yourself familiar with [Bots Architecture](bots-architecture/).
```

Comment:
a few simple diagrams and accompanying short description will REALLY HELP at this point.

External BOT architecture only.

Some attention should be paid to, in the architecture, -->  HOW DOES A BOT scale?   For example how can a BOT handle 1000 users concurrently.

Reply:
Let's leave this part as is at the moment. The architecture section will be reworked in the next PR